cmd/clusterctl/client/cluster/mover_test.go

```diff
@@ -788,3 +788,86 @@ func Test_objectsMoverService_checkTargetProviders(t *testing.T) {
 		})
 	}
 }
+
+func Test_objectMoverService_ensureNamespace(t *testing.T) {
+	type fields struct {
+		fromProxy Proxy
+	}
+	type args struct {
+		toProxy   Proxy
+		namespace string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "ensureNamespace doesn't fail given an existing namespace",
+			// Create a fake local cluster in namespace-1
+			fields: fields{getObjectGraphWithObjs(test.NewFakeCluster("namespace-1", "cl1").Objs()).proxy},
+			args: args{
+				// Create a fake cluster target with namespace-1 already existing
+				toProxy: getFakeProxyWithCRDs().WithObjs(test.NewFakeCluster("namespace-1", "cl1").Objs()...),
+				// Ensure namespace-1 gets created
+				namespace: "namespace-1",
+			},
+			wantErr: false,
+		},
+		{
+			name:   "ensureNamespace doesn't fail if the namespace does not already exist in the target",
+			fields: fields{getObjectGraphWithObjs(test.NewFakeCluster("namespace-1", "cl1").Objs()).proxy},
+			args: args{
+				// Create a fake cluster target with namespace-1 missing, but namespace-2 existing
+				toProxy: getFakeProxyWithCRDs().WithObjs(test.NewFakeCluster("namespace-1", "cl1").Objs()...),
+				// Ensure namespace-2 gets created
+				namespace: "namespace-2",
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			mover := objectMover{
+				fromProxy: tt.fields.fromProxy,
+			}
+
+			err := mover.ensureNamespace(tt.args.toProxy, tt.args.namespace)
+
+			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
+				return
+			}
+
+			g.Expect(err).NotTo(HaveOccurred())
+
+			// Check that the namespaces either existed or were created in the
+			// target.
+			csTo, err := tt.args.toProxy.NewClient()
+
+			if tt.wantErr {
```

Comment:
This shoud be `g.Expect(err).ToNot(HaveOccurred())`; `tt.wantErr` is related to the `ensureNamespace` call, not to `NewClient`